pkg/api/clustermanager.go

```diff
@@ -0,0 +1,153 @@
+package api
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// SyncSetList represents a list of SyncSets for a given cluster.
+type SyncSetList struct {
+	SyncSets []*SyncSet `json:"value"`
+	// The link used to get the next page of operations.
+	NextLink string `json:"nextLink,omitempty"`
+}
+
+type ClusterManagerConfigurationList struct {
+	ClusterManagerConfigurations []*ClusterManagerConfiguration `json:"value"`
+
+	NextLink string `json:"nextLink,omitempty"`
+}
+
+// ClusterManagerConfiguration represents the configuration from OpenShift Cluster Manager (OCM)
+type ClusterManagerConfiguration struct {
+	MissingFields
+
+	// ID is the unique identifier for the cluster manager configuration
+	ID                string                                `json:"id,omitempty"`
+	Name              string                                `json:"name,omitempty"`
+	ClusterResourceID string                                `json:"clusterResourceId,omitempty"`
+	Properties        ClusterManagerConfigurationProperties `json:"properties,omitempty"`
+	SystemData        *SystemData                           `json:"systemData,omitempty"`
+}
+
+// ClusterManagerConfigurationProperties houses the payloads the frontend receives for all OCM resources
+// we store them as a byte slice in cosmos
+type ClusterManagerConfigurationProperties struct {
+	Resources []byte `json:"resources,omitempty"`
+}
+
+// SyncSet represents a SyncSet for an Azure Red Hat OpenShift Cluster.
+type SyncSet struct {
+	// Required resource properties in ARM
+	ID   string `json:"id,omitempty" mutable:"case"`
+	Name string `json:"name,omitempty" mutable:"case"`
+	Type string `json:"type,omitempty" mutable:"case"`
```

Comment:
mutable should be only defined within the api definition, not the internal struct.